src/routes/email/email.controller.ts

```diff
@@ -0,0 +1,36 @@
+import { Body, Controller, Param, Post, UseGuards } from '@nestjs/common';
+import { EmailService } from '@/routes/email/email.service';
+import { EmailRegistrationGuard } from '@/routes/email/guards/email-registration.guard';
+import { TimestampGuard } from '@/routes/email/guards/timestamp.guard';
+import { OnlySafeOwnerGuard } from '@/routes/email/guards/only-safe-owner.guard';
+import { SaveEmailDto } from '@/routes/email/entities/save-email-dto.entity';
+import { ApiExcludeController, ApiTags } from '@nestjs/swagger';
+
+@ApiTags('email')
+@Controller({
+  path: 'chains/:chainId/safes/:safeAddress/emails',
+  version: '1',
+})
+@ApiExcludeController()
+export class EmailController {
+  constructor(private readonly service: EmailService) {}
+
+  @Post('')
+  @UseGuards(
+    EmailRegistrationGuard,
+    TimestampGuard(5 * 60 * 1000), // 5 minutes
```

Comment:
The `TimestampGuard` takes a number as an argument but since this is resolved on a build stage, passing the `configuration` value here is not possible (because the configuration is not resolved at this stage).

We can potentially change the `TimestampGuard` to accept a `() => number` instead of `number` so that it is lazily evaluated.